### PR TITLE
tests/e2e/test_pv_scale_ocs_create_delete_pvcs.py -- Enhanced

### PR DESCRIPTION
**Pull request by ramkiperiy**
Updated TC to skip memory leak function due to bz 1750328
Removed TODO for creating RAW block pod creation
Removed executor and implemented threads for parallel execution of function
Removed self variables which doesn't require across Class

Signed-off-by: ramkiperiy <rperiyas@redhat.com>

### CODE REVIEW

**Comment by ramkiperiy** [comment]
Test local run result pastebin link: https://paste.fedoraproject.org/paste/yPxNrDr~SXEIuyum9b7sYQ

File: tests/e2e/test_pv_scale_ocs_create_delete_pvcs.py [hunk 1]
```diff
@@ -138,41 +152,44 @@ def storageclass(self, storageclass_factory):
...
+        scale_pod_count = 120
+        size = '10Gi'
         test_run_time = 180
         self.all_pvc_obj, self.all_pod_obj = ([] for i in range(2))
         self.delete_pod_count = 0
 
         # Identify median memory value for each worker node
-        median_dict = helpers.get_memory_leak_median_value()
-        log.info(f"Median dict values for memory leak {median_dict}")
+        # TODO: Skipping memory leak median calculate because of bz 1750328
```
Comment (RazTamir) [inline tests/e2e/test_pv_scale_ocs_create_delete_pvcs.py]: do we have an issue to track this TODO?

File: tests/e2e/test_pv_scale_ocs_create_delete_pvcs.py [hunk 2]
```diff
@@ -138,41 +152,44 @@ def storageclass(self, storageclass_factory):
...
-                with ThreadPoolExecutor() as executor:
-                    log.info(f"Create {self.pvc_count_each_itr} and "
-                             f"in parallel delete {self.delete_pod_count} "
-                             f"pods & pvc")
-                    thread_list = [self.delete_pvc_pod(), self.create_pvc_pod(
-                        self.rbd_sc_obj, self.cephfs_sc_obj, self.pvc_count_each_itr,
-                        self.size, start_io)]
-                    for thread in thread_list:
-                        executor.submit(thread)
+                log.info(f"Create {pvc_count_each_itr} and "
```
Comment (RazTamir) [inline tests/e2e/test_pv_scale_ocs_create_delete_pvcs.py]:
```
log.info(
    f"Create {pvc_count_each_itr} and in parallel delete {self.delete_pod_count}"
    "pods & pvc"
)
```

File: tests/e2e/test_pv_scale_ocs_create_delete_pvcs.py [hunk 3]
```diff
@@ -138,41 +152,44 @@ def storageclass(self, storageclass_factory):
...
+                         f"in parallel delete {self.delete_pod_count} "
+                         f"pods & pvc")
+                thread1 = threading.Thread(target=self.delete_pvc_pod, args=())
+                thread2 = threading.Thread(target=self.create_pvc_pod, args=(
+                    self.rbd_sc_obj, self.cephfs_sc_obj, pvc_count_each_itr, size, start_io))
```
Comment (RazTamir) [inline tests/e2e/test_pv_scale_ocs_create_delete_pvcs.py]: ) in new line

File: tests/e2e/test_pv_scale_ocs_create_delete_pvcs.py (hunk 1 above)
Comment (ramkiperiy) [reply tests/e2e/test_pv_scale_ocs_create_delete_pvcs.py]: If bz 1750328 is resolved then this function works as expected, so there is no separate bug. But requires following enhancements.
1. For now memory leak function runs on all workers but in future we will have worker nodes for OCS and app pods, so function needs enhancement to collect data only on OCS nodes.
2. If a worker nodes has 3 OSD's for now function collects all the ceph-osd daemon stats but needs enhancement in analysis, may be based on process id, but should be intelligent to identify restart of process id(which will change).

File: tests/e2e/test_pv_scale_ocs_create_delete_pvcs.py (hunk 2 above)
Comment (ramkiperiy) [reply tests/e2e/test_pv_scale_ocs_create_delete_pvcs.py]: ack

File: tests/e2e/test_pv_scale_ocs_create_delete_pvcs.py (hunk 3 above)
Comment (ramkiperiy) [reply tests/e2e/test_pv_scale_ocs_create_delete_pvcs.py]: ack

File: tests/e2e/test_pv_scale_ocs_create_delete_pvcs.py (hunk 1 above)
Comment (ramkiperiy) [reply tests/e2e/test_pv_scale_ocs_create_delete_pvcs.py]: github issue: https://github.com/red-hat-storage/ocs-ci/issues/811

**Comment by ramkiperiy** [comment]
@vasukulkarni @RazTamir @ebenahar Comments are addressed, if the PR looks in good shape please approve

**Comment by vasukulkarni** [comment]
LGTM

**Pull request merged**